datafusion_util/src/lib.rs

```diff
@@ -347,6 +348,58 @@ pub fn context_with_table(batch: RecordBatch) -> SessionContext {
     ctx
 }
 
+/// Extract the column name from a an expression that might be treated with special NULL handling.
+///
+/// Normally we would just use the column itself:
+///
+/// ```text
+/// #col
+/// ```
+///
+/// However for certain interfaces we need to handle NULL values in a special way, e.g.:
+///
+/// ```text
+/// CASE WHEN #col IS NULL THEN <null placeholder> ELSE #col END
+/// ```
+///
+/// where `<null placeholder>` depends on the type, e.g. for string column this is an empty column.
+pub fn extract_null_wrapped_column(expr: &Expr) -> Option<String> {
+    match expr {
+        Expr::Column(column) => Some(column.name.clone()),
+        Expr::Case {
+            expr: None,
+            when_then_expr,
+            else_expr: Some(else_expr),
+        } if when_then_expr.len() == 1 => {
+            match (
+                when_then_expr[0].0.deref(),
+                when_then_expr[0].1.deref(),
+                else_expr.deref(),
+            ) {
+                (Expr::IsNull(column_expr), Expr::Literal(scalar), Expr::Column(column))
+                    if is_null_scalar(scalar)
+                        && (column_expr.deref() == &Expr::Column(column.clone())) =>
+                {
+                    Some(column.name.clone())
+                }
+                _ => None,
+            }
+        }
+        _ => None,
+    }
+}
+
+/// Check if a scalar looks like a default value for NULL values.
+fn is_null_scalar(scalar: &ScalarValue) -> bool {
```

Comment:
A more specific name might be:

```suggestion
fn is_influxrpc_default_null_scalar(scalar: &ScalarValue) -> bool {
```